dotCMS/src/integration-test/java/com/dotmarketing/webdav/DotWebdavHelperTest.java

```diff
@@ -0,0 +1,118 @@
+        final Resource folderResource = helper.getResourceFromURL(folderPath);
+        Assert.assertNotNull(folderResource);
+        Assert.assertTrue(folderResource instanceof FolderResource);
+        final String fileResourcePath = String.format("http://localhost:8080/webdav/live/1/%s/%s/%s/%s",host.getName(),parent.getName(),child.getName(),file.getName());
+        final Resource fileResource = helper.getResourceFromURL(fileResourcePath);
+        Assert.assertNotNull(fileResource);
+        Assert.assertTrue(fileResource instanceof FileResource);
+    }
+
+
+    @Test
+    public void Test_Get_Folder_Resource_Then_Get_File_Resource_Shuffled_Casing() throws IOException, DotDataException, DotSecurityException {
+        final SiteDataGen siteDataGen = new SiteDataGen();
+        final FolderDataGen folderDataGen = new FolderDataGen();
+        final Host host = siteDataGen.nextPersisted();
+        final Folder parent = folderDataGen.site(host).nextPersisted();
+        final Folder child = folderDataGen.parent(parent).nextPersisted();
+        java.io.File file = java.io.File.createTempFile("texto", ".txt");
+        FileUtil.write(file, "helloworld");
+        FileAssetDataGen fileAssetDataGen = new FileAssetDataGen(child, file);
+        fileAssetDataGen.nextPersisted();
+        final String folderPath = upperCaseRandom(String.format("http://localhost:8080/webdav/live/1/%s/%s/",host.getName(),parent.getName()), 8);
+
+        final Resource folderResource = helper.getResourceFromURL(folderPath);
+        Assert.assertNotNull(folderResource);
+        Assert.assertTrue(folderResource instanceof FolderResource);
+        final String fileResourcePath = upperCaseRandom(String.format("http://localhost:8080/webdav/live/1/%s/%s/%s/%s",host.getName(),parent.getName(),child.getName(),file.getName()),8);
+        final Resource fileResource = helper.getResourceFromURL(fileResourcePath);
+        Assert.assertNotNull(fileResource);
+        Assert.assertTrue(fileResource instanceof FileResource);
+    }
+
+    @Test
+    public void Test_Get_Folder_Resource_For_Non_Existing_Path() throws IOException, DotDataException, DotSecurityException {
+        String path = "http://localhost:8080/webdav/live/1/demo.dotcms.com/images/black.png";
+        final Resource folderResource = helper.getResourceFromURL(path);
+        Assert.assertNull(folderResource);
+    }
+
+    @Test
+    public void Test_Same_Resource_For_Paths_With_Different_Casing() throws IOException, DotDataException, DotSecurityException {
+        final SiteDataGen siteDataGen = new SiteDataGen();
+        final FolderDataGen folderDataGen = new FolderDataGen();
+        final Host host = siteDataGen.nextPersisted();
+        final Folder parent = folderDataGen.site(host).nextPersisted();
+        final Folder child = folderDataGen.parent(parent).nextPersisted();
+        java.io.File file = java.io.File.createTempFile("texto", ".txt");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'file' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4792071)